test/models/transforms/test_outcome.py

```diff
@@ -582,3 +583,120 @@ def test_power(self, seed=0):
             Y_tf_subset, Yvar_tf_subset = tf_subset(Y[..., [0]], None)
             self.assertTrue(torch.equal(Y_tf_subset, Y_tf[..., [0]]))
             self.assertIsNone(Yvar_tf_subset)
+
+    def test_bilog(self, seed=0):
+        torch.random.manual_seed(seed)
+
+        ms = (1, 2)
+        batch_shapes = (torch.Size(), torch.Size([2]))
+        dtypes = (torch.float, torch.double)
+
+        # test transform and untransform
+        for m, batch_shape, dtype in itertools.product(ms, batch_shapes, dtypes):
+
```

Comment:
We often define `tkwargs = {"device": self.device, "dtype": dtype}` and then you can just use `**tkwargs` each time you create a new tensor instead of explicitly passing in the device and dtype each time.